libdrgn/memory_reader.c

```diff
@@ -284,3 +284,33 @@ struct drgn_error *drgn_read_memory_file(void *buf, uint64_t address,
 	memset(p, 0, count);
 	return NULL;
 }
+
+struct drgn_error *drgn_read_memory_mmap(void *buf, uint64_t address,
+					 size_t count, uint64_t offset,
+					 void *arg, bool physical)
+{
+	struct drgn_memory_file_segment *file_segment = arg;
+	char *p = buf;
+	char *map_address = file_segment->map_address;
+	size_t file_count;
+
+	if (!map_address)
```

Comment:
We only use this callback if we successfully mapped this segment, right? So we should either remove this check or get rid of `drgn_read_memory_mmap()` and have `drgn_read_memory_file()` handle both the mmap and read cases.